app/views/products/_display_option.html.erb

```diff
@@ -1,10 +1,6 @@
 <%= form_tag request.fullpath, id: "display_option", class: "navbar-form", method: :get, style: "text-align:center;" do %>
-  <%= hidden_field_tag :slider_min, @slider[:min] %>
-  <%= hidden_field_tag :slider_max, @slider[:max] %>
-  <%= hidden_field_tag :slider_from, slider_from %>
-  <%= hidden_field_tag :slider_to, slider_to %>
   <div class="form-group">
-    <%= text_field_tag "keyword", params[:keyword] || nil, placeholder: t(:search), class: "form-control", style: "width: 35vw;" %>
+    <%= text_field_tag "q[name_cont_any]", params[:q]&.[](:name_cont_any), placeholder: t(:search), class: "form-control", style: "width: 35vw;" %>
```

Comment:
Tham khảo thêm search_form_for và search_field ransack có hỗ trợ e nhé. Để get value lúc search luôn hạn chế dùng  text_field_tag còn phải check default value
https://github.com/activerecord-hackery/ransack#ransacks-search_form_for-helper-replaces-form_for-for-creating-the-view-search-form

Reply:
Dạ